src/main/java/pl/javastart/task/ServiceUtility.java

```diff
@@ -0,0 +1,106 @@
+package pl.javastart.task;
+
+import java.io.*;
+import java.util.LinkedList;
+import java.util.Queue;
+import java.util.Scanner;
+
+public class ServiceUtility {
+    public static final int EXIT = 0;
+    public static final int ADD_CAR = 1;
+    public static final int POLL_CAR = 2;
+    private Queue<Car> cars = new LinkedList<>();
+    Scanner scanner = new Scanner(System.in);
+
+    public Queue<Car> getCarsQueueFromFile(String fileName) throws FileNotFoundException {
+        try (Scanner scanner = new Scanner(new File(fileName))) {
+            while (scanner.hasNextLine()) {
+                String line = scanner.nextLine();
+                String[] split = line.split(";");
+                String type = split[0];
+                String make = split[1];
+                String model = split[2];
+                int productionYear = Integer.parseInt(split[3]);
+                int mileage = Integer.parseInt(split[4]);
+                String vin = split[5];
+                Car car = new Car(type, make, model, productionYear, mileage, vin);
+                cars.add(car);
+            }
+        }
+        return cars;
+    }
+
+    public void run(String fileName) throws IOException {
+        File file = new File(fileName);
+        cars = getCarsQueueFromFile(fileName);
+        int option;
+        do {
+            option = chooseOption();
+            switch (option) {
+                case EXIT:
+                    if (!file.exists()) {
+                        file.createNewFile();
+                    }
+                    if (!cars.isEmpty()) {
+                        saveToFile(fileName, cars);
+                    }
+                    System.out.println("Program zakończony.");
+                    break;
+                case ADD_CAR:
+                    cars.add(createCar());
+                    break;
+                case POLL_CAR:
+                    if (!cars.isEmpty()) {
+                        System.out.println(cars.poll() + " w trakcie przeglądu.");
+                    } else {
+                        System.out.println("Brak pojazdów w kolejce");
+                    }
+                    break;
```

Comment:
Jeżeli w blokach case pojawia Ci się więcej logiki, to dobrze jest wydzielać to do pomocniczych metod, żeby zredukować długość takiej pojedynczej metody, tzn.:
```
case EXIT:
    exit();
    break;
case ADD_CAR:
...
```
u Ciebie wygląda to dobrze, ale przez takie wydzielenie metody zawsze to jeszcze trochę podniesie czytelność